src/SignedOwnershipProof.sol

```diff
@@ -19,42 +20,51 @@ contract SignedOwnershipProof {
 
     /// @notice Construct message that _ownershipSigner must sign as ownership proof
     /// @dev The RPC server uses this view function to create the ownership proof
-    /// @param nftOwner the address that currently owns the L1 NFT
+    /// @param signer the address that currently owns the L1 NFT
+    /// @param authorizer the address that currently owns the L1 NFT
     /// @param nonce the meta-transaction nonce for account
+    /// @param nftChainId the tokenId from nftContract for the NFT being utilized
     /// @param nftContract the mainnet contract address for the NFT being utilized
     /// @param tokenId the tokenId from nftContract for the NFT being utilized
+    /// @param timestamp the tokenId from nftContract for the NFT being utilized
     /// @return the message _ownershipSigner should sign
     function createMessage(
-        address nftOwner,
+        address signer,
+        address authorizer,
         uint256 nonce,
+        uint256 nftChainId,
         address nftContract,
         uint256 tokenId,
-        uint256 tokenNonce
-    ) public pure returns (bytes32) {
-        return keccak256(abi.encode(nftOwner, nonce, nftContract, tokenId, tokenNonce));
+        uint256 timestamp
+    ) public view returns (bytes32) {
+        return
+            keccak256(
+                abi.encode(signer, authorizer, nonce, nftChainId, nftContract, tokenId, block.chainid, timestamp)
+            );
     }
 
     /// @notice Verify signed OffchainLookup proof against meta-tx request data
     /// @dev Ensures that _ownershipSigner signed a message containing (nftOwner OR authorized address, nonce, nftContract, tokenId)
     /// @param req structured data submitted by EOA making a meta-transaction request
     /// @param signature the signature proof created by the ownership signer EOA
-    function verifyOwnershipProof(IForwardRequest.ERC721ForwardRequest memory req, bytes memory signature)
-        public
-        view
-        returns (bool)
-    {
-        // Verifies that ownership proof signature is signed by _ownerShip signer
-        // and that the embedded owner has authorized req.from
-        //
-        // Separately we must verify that the meta-tx signature also matches req.from
-        // and is signed by the EOA making the meta-transaction request.
-
-        IForwardRequest.PlaySession memory ps = _sessions[req.authorizer];
-        require(block.timestamp < ps.expiresAt, "Session Expired");
-        require(ps.authorized == req.from, "Signer not authorized");
+    function verifyOwnershipProof(
+        IForwardRequest.ERC721ForwardRequest memory req,
+        bytes memory signature,
+        uint256 timestamp
+    ) public view returns (bool) {
+        // TODO: what are the drift requirements here?
```

Comment:
I'd recommend going higher on this. There's not much risk to allowing some drift, and it'd be annoying and very "shitty web3 UI" for delays to make them have to resubmit without understanding why. I'd say it's worth going a few minutes to be safe.